Clean up public interface as much as possible

jameslamb opened the pull request:
In this PR, I tried to clean up a bunch of unnecessary stuff lingering in the package and, per our discussion today, moved as many things as I could inside the package. As a reminder, this exercise is important before our first release. The more we keep internal to the package now, the more flexibility we have to change `pkgnet` in the future.

The internals are still pretty tough to navigate right now, but we can work on that post-release since those changes wouldn't be user-facing.

This PR touches a lot of files, so I want to call out a few notable changes:
- hardcoded defaults for plotting options `layoutType` and `orphanNodeClusteringThreshold`
- got rid of confusing `reportCache` and `defaultCache` stuff
- changed most methods to `return(invisible(NULL))` instead of return random list objects
- removed `parse_extract_args` entirely
- removed tests on the expect names of private fields and methods. We should not be testing that. Tests are supposed to catch user-facing changes. Testing that a class has an exact set of private methods is an unnecessary administrative burden
- moved arguments about dependency types up out of `extract_network` and into `PackageFunctionReporter`. This should be the default way that we pass arguments through to key methods like that

I did not touch much documentation yet, know this PR still needs that. Have fun reviewing!

jameslamb commented:
oh also I eliminated `get_raw_data()`. Exposing a public getter that serves you the entire public namespace doesn't make sense to me

bburns632 commented on tests/testthat/test-PackageDependencyReporter.R:
```diff
@@ -62,54 +44,33 @@ test_that('PackageDependencyReporter Methods Work', {
-               , ignore.order = FALSE # enforcing this convention
-               , ignore.case = FALSE
+  expect_silent({
+      testObj$.__enclos_env__$private$extract_network()
```
This is new.  Cool.

bburns632 commented on R/AbstractPackageReporter.R:
```diff
@@ -81,55 +81,37 @@ AbstractPackageReporter <- R6::R6Class(
-            stop("get_summary_view has not been implemented.")
-        },
-
-        get_raw_data = function(){
```
Ohhh,  I didn't realize this was a thing we created.  yeah, good call on removing it.

bburns632 commented on R/CreatePackageReport.R:
```diff
@@ -68,52 +86,24 @@ CreatePackageReport <- function(packageName
+          X = packageReporters
+          , FUN = function(reporter){
+              reporter$set_package(packageName, packagePath)
+              return(reporter)
```
I believe you also need to run `reporter$calculate_all_metrics()` to populate each reporter.

jameslamb replied:
ok ok I take it back, you were right. Needed to add a call to actually calculate the metrics. We still do not need a public function called `calculate_all_metrics`, though. In my most recent commit, I actually implemented the thing I THOUGHT I'd done as of that last comment

jameslamb commented:
@bburns632 per your request, I went through the diff again and updated the list of noteworthy changes here. Note that I consider all of these to actually be within the scope of "internalizing things":


- hardcoded defaults for plotting options layoutType and orphanNodeClusteringThreshold
- got rid of confusing reportCache and defaultCache stuff
- changed most methods to return(invisible(NULL)) instead of return random list objects
- removed `calculate_all_metrics` entirely
- removed parse_extract_args entirely
- removed tests on the expect names of private fields and methods. We should not be testing that. Tests are supposed to catch user-facing changes. Testing that a class has an exact set of private methods is an unnecessary administrative burden
- moved arguments about dependency types up out of extract_network and into `PackageFunctionReporter` constructor. This should be the default way that we pass arguments through to key methods like that
- removed `get_package_name` function and instead added `package_name` active binding
- removed `get_report_markdown_path` and added `report_markdown_path` active binding
- moved some checking up out of `.BuildPackageReporters` and back into the body of `CreatePackageReport`. Pushing those checks into `.BuildPackageReporters` was not appropriate and made that function harder to understand
- Removed unnecessary `plot_network` calls in `.BuildPackageReporters`
- Simplified some cases where we were extracting a field from the object and then immediately reusing it only once
- added `calculate_test_coverage()` call to `extract_network` inside `PackageFunctionReporter`

jameslamb commented:
I just pushed new changes that update the documentation to reflect these changes. I also fixed the bug (good catch @bburns632 ) where I was missing calculation of node-level metrics.

also re-built the pkgdown site

bburns632 commented:
> moved some checking up out of .BuildPackageReporters and back into the body of CreatePackageReport. Pushing those checks into .BuildPackageReporters was not appropriate and made that function harder to understand

I disagree with this but not enough to protest this PR.  I think where you build the reporters is exactly where you should do any prebuild checks. 

> removed tests on the expect names of private fields and methods. 

Yeah, it was a little much.  I was just in the zone with unit test creation that day.

bburns632 commented:
There are some weird graph flipping with plot_network in function reporter.  Related to #54 and #56 but not the same.  I will create a new issues and slack video.  

Probably out of the scope of this PR but FYI

jameslamb commented on inst/package_report/package_summary_reporter.Rmd:
````diff
@@ -7,15 +7,7 @@ params:
-Number of objects
-
-```{r echo=FALSE}
-reporter$get_objects()
````
oh true! yeah I don't think that ever hit master

jameslamb commented:
@bburns632 I moved those checks up out of `.BuiltPackageReporters` to make it a bit clearer what is happening inside that function. Like IMHO it's easier to read without the checks

p.s. (I'm annoyed that you can't comment on comments all the time in GitHub)

jameslamb commented:
need to look again at the `graphviz` vs. `plot_network` thing. I think there was a reason we needed active binding for `graphviz` and unfortunately active binding must be public

jameslamb commented:
@bburns632 yep you were right! I moved `plot_network` inside the package. Nice!

bburns632 commented:
Change two unit tests and you're g2g

jameslamb commented:
Unit tests pushed! will merge whenever travis builds 🤞
